var/spack/repos/builtin/packages/bash/package.py

```diff
@@ -29,7 +29,7 @@ class Bash(Package):
     """The GNU Project's Bourne Again SHell."""
 
     homepage = "https://www.gnu.org/software/bash/"
-    url      = "ftp://ftp.gnu.org/gnu/bash/bash-4.3.tar.gz"
+    url      = "https://ftp.gnu.org/gnu/bash/bash-4.3.tar.gz"
```

Comment:
Not all sites support `https` yet. I prefer `https` over `http` when possible. 

However, to simplify the implemetation of upcomming ftp-http test functions in pakcage sanity checks, retrieval URLs are categorized into 1) `ftp` and 2) `http` or `https`. In this case, only one `http_site_list` is needed. Both `http` and `https` will pass this test.